### PR TITLE
Add Aim logging to progress_bar

### PR DESCRIPTION
**Pull request by alberttorosyan**
**Patch Description**
- Added `AimProgressBarWrapper` class; an implementation of `BaseProgressBar` using Aim, an open-source experiment tracker: https://github.com/aimhubio/aim.
- Added two new arguments to `CommonConfig` used to control `AimProgressBarWrapper` creation/usage:
  - `aim_repo`: path to the Aim repository.
  - `aim_run_hash`: If specified, will be used to determine `aim.Run` for metadata tracking. If skipped, `checkpoint.save_dir` will be used to find and re-open the existing `Run` or if no matching `Run` is found, new one is created.

**Testing steps**
Make sure `progress_bar` logs data to selected outputs, with Aim logger enabled/disabled.
Run unit-tests.

- [ ] Was this discussed/approved via a Github issue?
- [x] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: metaseq/dataclass/configs.py
Comment (suchenzang) [inline metaseq/dataclass/configs.py:109]: Why should this pathing be complete separate from where all other logs go?

File: metaseq/logging/progress_bar.py
Comment (suchenzang) [inline metaseq/logging/progress_bar.py:407]: Please add corresponding installation requirement to setup.py.

**Comment by suchenzang** [comment]
This raises a good todo to tackle before we start adding an excessive amount of args / flags here: https://github.com/facebookresearch/metaseq/issues/93

File: metaseq/dataclass/configs.py
Comment (alberttorosyan) [reply metaseq/dataclass/configs.py:109]: @suchenzang Aim provides remote tracking functionality, and `aim_repo` can refer a remote server URI rather than local path.

**Comment by gorarakelyan** [comment]
@suchenzang thanks a lot for the review 🙌 
I wonder where we should add a mini-guide on how to use Aim? To [docs/training](https://github.com/facebookresearch/metaseq/blob/main/docs/training.md) or to README?

File: metaseq/logging/progress_bar.py
Comment (alberttorosyan) [reply metaseq/logging/progress_bar.py:407]: sure, will push changes shortly

Comment (alberttorosyan) [reply metaseq/logging/progress_bar.py:407]: done ✅

**Comment by suchenzang** [comment]
Sorry for the delay in getting back - adding a mini-guide to using Aim would be much appreciated! Yes - feel free to add to https://github.com/facebookresearch/metaseq/blob/main/docs/training.md !

I've been blocking on this PR due to https://github.com/facebookresearch/metaseq/pull/96 - once that goes through, will request a modification to this PR accordingly as well.

**Comment by suchenzang** [comment]
@gorarakelyan You may need to sign the CLA check (not sure if it automatically shows up for you here since @alberttorosyan opened the PR, but going to https://code.facebook.com/cla should work) so that we can merge this.  Ping me when it's ready for review!

**Comment by gorarakelyan** [comment]
@suchenzang done ✅

**Review by suchenzang — APPROVED** [review]
Cool! Can't wait to try this out!

**Comment by suchenzang** [comment]
@gorarakelyan one more nit: before merging, can you make sure our lint test passes?

**Comment by alberttorosyan** [comment]
@suchenzang just pushed changes with linter issues fix. but the `circleci` tests seemed to stuck with no progress, so I cancelled them. is there any pre-requisite for running these tests other than mentioned in `CONTRIBUTING.md`?

**Comment by alberttorosyan** [comment]
> @suchenzang just pushed changes with linter issues fix. but the `circleci` tests seemed to stuck with no progress, so I cancelled them. is there any pre-requisite for running these tests other than mentioned in `CONTRIBUTING.md`?

Hey @suchenzang! Just wanted to double-check, should I take any additional actions here or we can proceed forward and merge PR?

**Comment by suchenzang** [comment]
@alberttorosyan Seems like I can't rerun the gpu tests either - should be good to merge though; will do now! Thanks for the PR!

**Pull request merged**
